From 58df476df123d2e0d7b2cb7b1009cf4b51b6d506 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 8 Jan 2022 19:42:34 +0100
Subject: [PATCH 007/177] Bromite package name

---
 chrome/android/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index b6656598a1..9de45f564a 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -43,7 +43,7 @@ if (android_64bit_target_cpu && skip_secondary_abi_for_cq) {
   assert(current_toolchain != android_secondary_abi_toolchain)
 }
 
-_default_package = "org.chromium.chrome"
+_default_package = "org.bromite.bromite"
 if (android_channel != "default" && android_channel != "stable") {
   # android-binary-size trybot may checks if an internal Chrome variant's
   # AndroidManifest is as expected by ensuring the differences between its
-- 
2.35.1

