From f191c8ff95a8b2cfdd7baa804ad7e5e407fbd485 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Thu, 12 Oct 2017 08:09:24 +0200
Subject: [PATCH 017/177] Do not hide component extensions

---
 extensions/common/extension.cc | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index 731da16dcc..6937ab98e1 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -469,9 +469,8 @@ bool Extension::ShouldDisplayInNewTabPage() const {
 }
 
 bool Extension::ShouldExposeViaManagementAPI() const {
-  // Hide component extensions because they are only extensions as an
-  // implementation detail of Chrome.
-  return !extensions::Manifest::IsComponentLocation(location());
+  // do not hide component extensions
+  return true;
 }
 
 Extension::ManifestData* Extension::GetManifestData(const std::string& key)
-- 
2.35.1

