From 11dfebcd6113ff95b1d69ef209b25fcb8e21e35d Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Tue, 26 Jan 2021 00:21:51 +0100
Subject: [PATCH 080/107] Ignore enterprise policies for secure DNS

---
 .../chrome/browser/enterprise/util/EnterpriseInfo.java        | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/chrome/browser/enterprise/util/android/java/src/org/chromium/chrome/browser/enterprise/util/EnterpriseInfo.java b/chrome/browser/enterprise/util/android/java/src/org/chromium/chrome/browser/enterprise/util/EnterpriseInfo.java
index e3ead444d1..7d2c7d7b12 100644
--- a/chrome/browser/enterprise/util/android/java/src/org/chromium/chrome/browser/enterprise/util/EnterpriseInfo.java
+++ b/chrome/browser/enterprise/util/android/java/src/org/chromium/chrome/browser/enterprise/util/EnterpriseInfo.java
@@ -91,6 +91,10 @@ public class EnterpriseInfo {
         ThreadUtils.assertOnUiThread();
         assert callback != null;
 
+        if (true) {
+           return;
+        }
+
         // If there is already a cached result post a task to return it.
         if (mOwnedState != null) {
             mHandler.post(() -> callback.onResult(mOwnedState));
-- 
2.35.1

