diff --git a/base/android/java/src/org/chromium/base/ImmutableFlagWithDefault.java b/base/android/java/src/org/chromium/base/ImmutableFlagWithDefault.java
new file mode 100644
index 0000000000000..39b4247404928
--- /dev/null
+++ b/base/android/java/src/org/chromium/base/ImmutableFlagWithDefault.java
@@ -0,0 +1,25 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.chromium.base;
+
+import org.chromium.build.annotations.NullMarked;
+import org.chromium.build.annotations.Nullable;
+
+@NullMarked
+public class ImmutableFlagWithDefault extends MutableFlagWithSafeDefault {
+    private final boolean mDefaultValue;
+    private @Nullable Boolean mInMemoryCachedValue;
+
+    public ImmutableFlagWithDefault(
+            FeatureMap featureMap, String featureName, boolean defaultValue) {
+        super(featureMap, featureName, defaultValue);
+        mDefaultValue = defaultValue;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return mDefaultValue;
+    }
+}
diff --git a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
index 52ca99ec51895..227ca19b04726 100644
--- a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
+++ b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
@@ -1271,7 +1271,7 @@ public abstract class ChromeFeatureList {
             newCachedFlag(WEB_UI_ANDROID_THEMING, false);
     // keep-sorted end
 
-    public static final List<CachedFlag> sFlagsCachedFullBrowser =
+    public static final List<CachedFlag> sFlagsCachedFullBrowserChromium =
             List.of(
                     // keep-sorted start
                     sAccountForSuppressedKeyboardInsets,
diff --git a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/include_all_directory.java b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/include_all_directory.java
new file mode 100644
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/include_all_directory.java
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/java_template/CromiteCachedFlag.java.tmpl b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/java_template/CromiteCachedFlag.java.tmpl
new file mode 100644
index 0000000000000..3355066539d9f
--- /dev/null
+++ b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/cromite/java_template/CromiteCachedFlag.java.tmpl
@@ -0,0 +1,47 @@
+/*
+    This file is part of Cromite.
+
+    Cromite is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Cromite is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Cromite. If not, see <https://www.gnu.org/licenses/>.
+
+*/
+
+{DESCRIPTION}
+
+package org.chromium.chrome.browser.flags.cromite;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.chromium.components.cached_flags.CachedFlag;
+
+{IMPORT_CLAUSE}
+
+public class CromiteCachedFlagImplBase {{
+
+    static protected final ArrayList<CachedFlag> mItemList;
+
+    private static void add(CachedFlag flag) {{
+        mItemList.add(flag);
+    }}
+
+    public static List<CachedFlag> getList() {{
+        return mItemList;
+    }}
+
+    static {{
+        mItemList = new ArrayList<CachedFlag>();
+{ADD_CLAUSE}
+    }}
+
+}}
diff --git a/chrome/browser/settings/android/java/src/org/chromium/chrome/browser/settings/ChromeBaseSettingsFragment.java b/chrome/browser/settings/android/java/src/org/chromium/chrome/browser/settings/ChromeBaseSettingsFragment.java
index cba533e44ebff..b11189d4bb409 100644
--- a/chrome/browser/settings/android/java/src/org/chromium/chrome/browser/settings/ChromeBaseSettingsFragment.java
+++ b/chrome/browser/settings/android/java/src/org/chromium/chrome/browser/settings/ChromeBaseSettingsFragment.java
@@ -10,7 +10,7 @@ import android.content.Context;
 import android.os.Bundle;
 import android.view.ContextThemeWrapper;
 import android.view.LayoutInflater;
-
+import java.util.function.Supplier;
 import androidx.annotation.StringRes;
 import androidx.preference.PreferenceFragmentCompat;
 
@@ -48,7 +48,48 @@ public abstract class ChromeBaseSettingsFragment extends PreferenceFragmentCompa
     private SettingsCustomTabLauncher mCustomTabLauncher;
     private @Nullable PreferenceUpdateObserver mPreferenceUpdateObserver;
     private @Nullable Context mThemedContext;
+    private Supplier<ChromeBaseSettingsFragment.RequireRestartDelegate> mRequireRestartDelegateSupplier;
+
+    public interface RequireRestartDelegate {
+        void RequireRestart();
+    }
+
+    public void setRequestRestartDelegateSupplier(
+                    Supplier<ChromeBaseSettingsFragment.RequireRestartDelegate> delegate) {
+        mRequireRestartDelegateSupplier = delegate;
+    }
 
+    public void onCreatePreferencesCromite(Bundle savedInstanceState, String rootKey) {
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        onCreatePreferencesCromite(savedInstanceState, rootKey);
+
+        PreferenceScreen prefScreen = getPreferenceScreen();
+        int prefCount = prefScreen.getPreferenceCount();
+
+        for(int i=0; i < prefCount; i++) {
+            Preference pref = prefScreen.getPreference(i);
+            if (pref instanceof ChromeSwitchPreference) {
+                ChromeSwitchPreference switchPref = (ChromeSwitchPreference)pref;
+                String featureName = switchPref.getFeatureName();
+                if (featureName == null)
+                    continue;
+
+                boolean enabled = CromiteNativeUtils.isFlagEnabled(featureName);
+                switchPref.setChecked(enabled);
+
+                switchPref.setOnPreferenceChangeListener((preference, newValue) -> {
+                    CromiteNativeUtils.setFlagEnabled(featureName, (boolean)newValue);
+                    if (switchPref.needRestart()) {
+                        mRequireRestartDelegateSupplier.get().RequireRestart();
+                    }
+                    return true;
+                });
+            }
+        }
+    }
     @Override
     public void onAttach(Context context) {
         if (!SettingsInTab.isEnabled()) {
diff --git a/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/Add-cromite-flags-support.grdp b/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/Add-cromite-flags-support.grdp
new file mode 100644
index 0000000000000..df7c8ced94718
--- /dev/null
+++ b/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/Add-cromite-flags-support.grdp
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<grit-part>
+    <message name="IDS_RELAUNCH" desc="Relaunch message">
+        Relaunch
+    </message>
+    <message name="IDS_UI_RELAUNCH_NOTICE" desc="Summary for relaunch message">
+        Your changes will take effect the next time you relaunch Cromite.
+    </message>
+</grit-part>
diff --git a/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/placeholder.txt b/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/placeholder.txt
new file mode 100644
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/chrome/browser/ui/android/strings/cromite_android_chrome_strings_grd/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
index 05e8b37464557..7b9aa2689b507 100644
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -573,4 +573,5 @@ BASE_FEATURE(kOSCryptAsyncAvailabilityInfoBar,
 BASE_FEATURE(kDeferLayoutDuringBrowserStartup,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+#include "cromite_flags/chrome_browser_ui_ui_features_cc.inc"
 }  // namespace features
diff --git a/components/cromite_components_strings_grd/placeholder.txt b/components/cromite_components_strings_grd/placeholder.txt
new file mode 100644
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/components/cromite_components_strings_grd/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/components/webui/flags/resources/app.ts b/components/webui/flags/resources/app.ts
index fa5c95008844f..caf93f9116691 100644
--- a/components/webui/flags/resources/app.ts
+++ b/components/webui/flags/resources/app.ts
@@ -198,6 +198,8 @@ export class FlagsAppElement extends CrLitElement {
 
   protected accessor defaultFeatures: Feature[] = [];
   protected accessor nonDefaultFeatures: Feature[] = [];
+  protected defaultCromiteFeatures: Feature[] = [];
+  protected nonDefaultCromiteFeatures: Feature[] = [];
   protected accessor searching: boolean = false;
   protected accessor needsRestart: boolean = false;
   protected accessor importError: string = '';
@@ -205,6 +207,7 @@ export class FlagsAppElement extends CrLitElement {
       loadTimeData.getBoolean('webuiRoundedIconsEnabled');
 
   private announceStatusDelayMs: number = 100;
+  private onlyCromiteFlags: boolean = false;
   private featuresResolver: PromiseResolver<void> = new PromiseResolver();
   private flagSearch: FlagSearch|null = null;
   private flagsRedirects_: Record<string, string> = FLAG_REDIRECTS;
diff --git a/cromite_flags/BUILD.gn b/cromite_flags/BUILD.gn
new file mode 100755
index 0000000000000..82b9b3e1a869c
--- /dev/null
+++ b/cromite_flags/BUILD.gn
@@ -0,0 +1,174 @@
+# This file is part of Bromite.
+
+# Bromite is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# Bromite is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with Bromite. If not, see <https://www.gnu.org/licenses/>.
+
+# for placeholder.txt:
+#
+# this file is intentionally empty
+#
+
+cpp_bromite_include("chrome_browser_about_flags_cc") {
+  inputs = [ "//cromite_flags/chrome/browser/about_flags_cc/placeholder.txt" ]
+  output_file = "chrome_browser_about_flags_cc.inc"
+}
+
+cpp_bromite_include("chrome_common_chrome_features_cc") {
+  inputs = [ "//cromite_flags/chrome/common/chrome_features_cc/placeholder.txt" ]
+  output_file = "chrome_common_chrome_features_cc.inc"
+}
+
+cpp_bromite_include("content_common_features_cc") {
+  inputs = [ "//cromite_flags/content/common/features_cc/placeholder.txt" ]
+  output_file = "content_common_features_cc.inc"
+}
+
+cpp_bromite_include("content_public_common_content_features_h") {
+  inputs = [ "//cromite_flags/content/public/common/content_features_h/placeholder.txt" ]
+  output_file = "content_public_common_content_features_h.inc"
+}
+
+cpp_bromite_include("content_public_common_content_features_cc") {
+  inputs = [ "//cromite_flags/content/public/common/content_features_cc/placeholder.txt" ]
+  output_file = "content_public_common_content_features_cc.inc"
+}
+
+cpp_bromite_include("third_party_blink_common_features_cc") {
+  inputs = [ "//cromite_flags/third_party/blink/common/features_cc/placeholder.txt" ]
+  output_file = "third_party_blink_common_features_cc.inc"
+}
+
+cpp_bromite_include("third_party_blink_common_features_h") {
+  inputs = [ "//cromite_flags/third_party/blink/common/features_h/placeholder.txt" ]
+  output_file = "third_party_blink_common_features_h.inc"
+}
+
+cpp_bromite_include("chrome_browser_flags_android_chrome_feature_list_cc") {
+  inputs = [ "//cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/placeholder.txt" ]
+  output_file = "chrome_browser_flags_android_chrome_feature_list_cc.inc"
+}
+
+cpp_bromite_include("chrome_browser_flags_android_chrome_feature_list_h") {
+  inputs = [ "//cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/placeholder.txt" ]
+  output_file = "chrome_browser_flags_android_chrome_feature_list_h.inc"
+}
+
+cpp_bromite_include("chrome_browser_browser_features_cc") {
+  inputs = [ "//cromite_flags/chrome/browser/browser_features_cc/placeholder.txt" ]
+  output_file = "chrome_browser_browser_features_cc.inc"
+}
+
+cpp_bromite_include("chrome_browser_browser_features_h") {
+  inputs = [ "//cromite_flags/chrome/browser/browser_features_h/placeholder.txt" ]
+  output_file = "chrome_browser_browser_features_h.inc"
+}
+
+cpp_bromite_include("chrome_browser_ui_ui_features_cc") {
+  inputs = [ "//cromite_flags/chrome/browser/ui/ui_features_cc/placeholder.txt" ]
+  output_file = "chrome_browser_ui_ui_features_cc.inc"
+}
+
+cpp_bromite_include("media_base_media_switches_cc") {
+  inputs = [ "//cromite_flags/media/base/media_switches_cc/placeholder.txt" ]
+  output_file = "media_base_media_switches_cc.inc"
+}
+
+cpp_bromite_include("media_base_media_switches_h") {
+  inputs = [ "//cromite_flags/media/base/media_switches_h/placeholder.txt" ]
+  output_file = "media_base_media_switches_h.inc"
+}
+
+cpp_bromite_include("components_content_settings_core_common_features_cc") {
+  inputs = [ "//cromite_flags/components/content_settings/core/common/features_cc/placeholder.txt" ]
+  output_file = "components_content_settings_core_common_features_cc.inc"
+}
+
+cpp_bromite_include("components_permissions_features_cc") {
+  inputs = [ "//cromite_flags/components/permissions/features_cc/placeholder.txt" ]
+  output_file = "components_permissions_features_cc.inc"
+}
+
+cpp_bromite_include("components_offline_pages_core_offline_page_feature_cc") {
+  inputs = [ "//cromite_flags/components/offline_pages/core/offline_page_feature_cc/placeholder.txt" ]
+  output_file = "components_offline_pages_core_offline_page_feature_cc.inc"
+}
+
+cpp_bromite_include("components_offline_pages_core_offline_page_feature_h") {
+  inputs = [ "//cromite_flags/components/offline_pages/core/offline_page_feature_h/placeholder.txt" ]
+  output_file = "components_offline_pages_core_offline_page_feature_h.inc"
+}
+
+cpp_bromite_include("net_base_features_cc") {
+  inputs = [ "//cromite_flags/net/base/features_cc/placeholder.txt" ]
+  output_file = "net_base_features_cc.inc"
+}
+
+cpp_bromite_include("net_base_features_h") {
+  inputs = [ "//cromite_flags/net/base/features_h/placeholder.txt" ]
+  output_file = "net_base_features_h.inc"
+}
+
+cpp_bromite_include("services_network_public_cpp_features_cc") {
+  inputs = [ "//cromite_flags/services/network/public/cpp/features_cc/placeholder.txt" ]
+  output_file = "services_network_public_cpp_features_cc.inc"
+}
+
+cpp_bromite_include("services_network_public_cpp_features_h") {
+  inputs = [ "//cromite_flags/services/network/public/cpp/features_h/placeholder.txt" ]
+  output_file = "services_network_public_cpp_features_h.inc"
+}
+
+cpp_bromite_include("ui_base_features_cc") {
+  inputs = [ "//cromite_flags/ui/base/features_cc/placeholder.txt" ]
+  output_file = "ui_base_features_cc.inc"
+}
+
+cpp_bromite_include("ui_base_features_h") {
+  inputs = [ "//cromite_flags/ui/base/features_h/placeholder.txt" ]
+  output_file = "ui_base_features_h.inc"
+}
+
+cpp_bromite_include("components_password_manager_core_browser_features_password_features_cc") {
+  inputs = [ "//cromite_flags/components/password_manager/core/browser/features/password_features_cc/placeholder.txt" ]
+  output_file = "components_password_manager_core_browser_features_password_features_cc.inc"
+}
+
+component("cromite_flags") {
+  deps = [
+    ":content_common_features_cc",
+    ":content_public_common_content_features_cc",
+    ":content_public_common_content_features_h",
+    ":components_content_settings_core_common_features_cc",
+    ":components_permissions_features_cc",
+    ":components_offline_pages_core_offline_page_feature_cc",
+    ":components_offline_pages_core_offline_page_feature_h",
+    ":components_password_manager_core_browser_features_password_features_cc",
+    ":media_base_media_switches_cc",
+    ":media_base_media_switches_h",
+    ":net_base_features_cc",
+    ":net_base_features_h",
+    ":chrome_common_chrome_features_cc",
+    ":chrome_browser_about_flags_cc",
+    ":chrome_browser_flags_android_chrome_feature_list_cc",
+    ":chrome_browser_flags_android_chrome_feature_list_h",
+    ":chrome_browser_ui_ui_features_cc",
+    ":chrome_browser_browser_features_cc",
+    ":chrome_browser_browser_features_h",
+    ":services_network_public_cpp_features_cc",
+    ":services_network_public_cpp_features_h",
+    ":third_party_blink_common_features_cc",
+    ":third_party_blink_common_features_h",
+    ":ui_base_features_cc",
+    ":ui_base_features_h",
+  ]
+}
diff --git a/cromite_flags/chrome/browser/about_flags_cc/placeholder.txt b/cromite_flags/chrome/browser/about_flags_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/about_flags_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/browser_features_cc/placeholder.txt b/cromite_flags/chrome/browser/browser_features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/browser_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/browser_features_h/placeholder.txt b/cromite_flags/chrome/browser/browser_features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/browser_features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/placeholder.txt b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/placeholder.txt b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/ui/ui_features_cc/placeholder.txt b/cromite_flags/chrome/browser/ui/ui_features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/browser/ui/ui_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/common/chrome_features_cc/placeholder.txt b/cromite_flags/chrome/common/chrome_features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/common/chrome_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/common/chrome_features_h/placeholder.txt b/cromite_flags/chrome/common/chrome_features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/chrome/common/chrome_features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/content_settings/core/common/features_cc/placeholder.txt b/cromite_flags/components/content_settings/core/common/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/components/content_settings/core/common/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/offline_pages/core/offline_page_feature_cc/placeholder.txt b/cromite_flags/components/offline_pages/core/offline_page_feature_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/components/offline_pages/core/offline_page_feature_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/offline_pages/core/offline_page_feature_h/placeholder.txt b/cromite_flags/components/offline_pages/core/offline_page_feature_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/components/offline_pages/core/offline_page_feature_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/password_manager/core/browser/features/password_features_cc/placeholder.txt b/cromite_flags/components/password_manager/core/browser/features/password_features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/components/password_manager/core/browser/features/password_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/permissions/features_cc/placeholder.txt b/cromite_flags/components/permissions/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/components/permissions/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/content/common/features_cc/placeholder.txt b/cromite_flags/content/common/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/content/common/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/content/public/common/content_features_cc/placeholder.txt b/cromite_flags/content/public/common/content_features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/content/public/common/content_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/content/public/common/content_features_h/placeholder.txt b/cromite_flags/content/public/common/content_features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/content/public/common/content_features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/media/base/media_switches_cc/placeholder.txt b/cromite_flags/media/base/media_switches_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/media/base/media_switches_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/media/base/media_switches_h/placeholder.txt b/cromite_flags/media/base/media_switches_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/media/base/media_switches_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/net/base/features_cc/placeholder.txt b/cromite_flags/net/base/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/net/base/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/net/base/features_h/placeholder.txt b/cromite_flags/net/base/features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/net/base/features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/services/network/public/cpp/features_cc/placeholder.txt b/cromite_flags/services/network/public/cpp/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/services/network/public/cpp/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/services/network/public/cpp/features_h/placeholder.txt b/cromite_flags/services/network/public/cpp/features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/services/network/public/cpp/features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/third_party/blink/common/features_cc/placeholder.txt b/cromite_flags/third_party/blink/common/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/third_party/blink/common/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/third_party/blink/common/features_h/placeholder.txt b/cromite_flags/third_party/blink/common/features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/third_party/blink/common/features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/ui/base/features_cc/placeholder.txt b/cromite_flags/ui/base/features_cc/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/ui/base/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/ui/base/features_h/placeholder.txt b/cromite_flags/ui/base/features_h/placeholder.txt
new file mode 100755
index 0000000000000..5174b09c2273b
--- /dev/null
+++ b/cromite_flags/ui/base/features_h/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/media/base/media_switches.h b/media/base/media_switches.h
index a510eb58b0fb6..53632072b4b63 100644
--- a/media/base/media_switches.h
+++ b/media/base/media_switches.h
@@ -664,4 +664,5 @@ MEDIA_EXPORT BASE_DECLARE_FEATURE(kRustMpegAudioDataParser);
 
 }  // namespace media
 
+#include "cromite_flags/media_base_media_switches_h.inc"
 #endif  // MEDIA_BASE_MEDIA_SWITCHES_H_
diff --git a/net/base/features.cc b/net/base/features.cc
index 54e646c9a6c7f..48aae217a2f7f 100644
--- a/net/base/features.cc
+++ b/net/base/features.cc
@@ -980,4 +980,5 @@ BASE_FEATURE(kEnableBackendCleanupTrackerOnHttpCache,
 BASE_FEATURE(kPartitionWebSocketEndpointLocksByNetworkAnonymizationKey,
              base::FEATURE_ENABLED_BY_DEFAULT);
 
+#include "cromite_flags/net_base_features_cc.inc"
 }  // namespace net::features
diff --git a/net/base/features.h b/net/base/features.h
index 238ae5d62b6f3..a4af1f4e9e191 100644
--- a/net/base/features.h
+++ b/net/base/features.h
@@ -1006,6 +1006,7 @@ NET_EXPORT BASE_DECLARE_FEATURE(kEnableBackendCleanupTrackerOnHttpCache);
 NET_EXPORT BASE_DECLARE_FEATURE(
     kPartitionWebSocketEndpointLocksByNetworkAnonymizationKey);
 
+#include "cromite_flags/net_base_features_h.inc"
 }  // namespace net::features
 
 #endif  // NET_BASE_FEATURES_H_
diff --git a/ui/base/ui_base_features.cc b/ui/base/ui_base_features.cc
index b0bc79e3094b6..b70953e326f37 100644
--- a/ui/base/ui_base_features.cc
+++ b/ui/base/ui_base_features.cc
@@ -542,4 +542,5 @@ bool IsWebUIRoundedIconsEnabled() {
   return base::FeatureList::IsEnabled(kWebUIRoundedIcons);
 }
 
+#include "cromite_flags/ui_base_features_cc.inc"
 }  // namespace features
diff --git a/ui/base/ui_base_features.h b/ui/base/ui_base_features.h
index b174726217a59..beaef84de8749 100644
--- a/ui/base/ui_base_features.h
+++ b/ui/base/ui_base_features.h
@@ -363,6 +363,7 @@ bool IsRoundedIconsEnabled();
 COMPONENT_EXPORT(UI_BASE_FEATURES)
 bool IsWebUIRoundedIconsEnabled();
 
+#include "cromite_flags/ui_base_features_h.inc"
 }  // namespace features
 
 #endif  // UI_BASE_UI_BASE_FEATURES_H_
