From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Sun, 25 Sep 2022 18:25:52 +0530
Subject: Build fix for "Remove signin and sync integrations"

---
 .../subscriptions/CommerceSubscriptionsServiceFactory.java       | 1 -
 1 file changed, 1 deletion(-)

diff --git a/chrome/browser/commerce/subscriptions/android/java/src/org/chromium/chrome/browser/subscriptions/CommerceSubscriptionsServiceFactory.java b/chrome/browser/commerce/subscriptions/android/java/src/org/chromium/chrome/browser/subscriptions/CommerceSubscriptionsServiceFactory.java
--- a/chrome/browser/commerce/subscriptions/android/java/src/org/chromium/chrome/browser/subscriptions/CommerceSubscriptionsServiceFactory.java
+++ b/chrome/browser/commerce/subscriptions/android/java/src/org/chromium/chrome/browser/subscriptions/CommerceSubscriptionsServiceFactory.java
@@ -68,7 +68,6 @@ public class CommerceSubscriptionsServiceFactory {
                     PriceDropNotificationManagerFactory.create();
             service = new CommerceSubscriptionsService(
                     new SubscriptionsManagerImpl(profile, priceDropNotificationManager),
-                    IdentityServicesProvider.get().getIdentityManager(profile),
                     priceDropNotificationManager);
             sProfileToSubscriptionsService.put(profile, service);
         }
-- 
2.37.3

