From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Deepa <deepa@calyx.org>
Date: Mon, 24 Aug 2026 23:36:13 +0000
Subject: do not hide .orig files

---
 .gitignore | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index e9d02975508c7..28c47973a3596 100644
--- a/.gitignore
+++ b/.gitignore
@@ -5,7 +5,7 @@
 *.mk
 *.ncb
 *.opensdf
-*.orig
+# *.orig
 *.pdb
 *.profraw
 *.props
-- 
2.39.5

