From: Michael Bestas <mkbestas@gmail.com>
Date: Mon, 22 Jul 2024 18:58:32 +0000
Subject: Fix search_engines build error

---
 chrome/browser/search_engines/BUILD.gn | 1 -
 1 file changed, 1 deletion(-)

diff --git a/chrome/browser/search_engines/BUILD.gn b/chrome/browser/search_engines/BUILD.gn
--- a/chrome/browser/search_engines/BUILD.gn
+++ b/chrome/browser/search_engines/BUILD.gn
@@ -32,7 +32,6 @@ source_set("search_engines") {
 
   deps = [
     "//chrome/browser:browser_process",
-    "//chrome/browser/google",
     "//chrome/common:channel_info",
     "//chrome/common:non_code_constants",
     "//components/google/core/common",
-- 

