From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 5 Oct 2023 20:55:10 +0000
Subject: fixup! Add build args, keep last / on top

---
 args.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/args.gn b/args.gn
--- a/args.gn
+++ b/args.gn
@@ -46,7 +46,7 @@ use_official_google_api_keys=false
 use_rtti=false
 use_stable_package_name_for_trichrome=false
 #use_sysroot=false
-webview_includes_weblayer=true
+webview_includes_weblayer=false
 
 # certs/chromium
 trichrome_certdigest = "d674c702c9415fb3b6ee9a1a80b8c45ad6ea5395379eb21f51122938c15f6be8"
-- 

