From: Tommy Webb <tommy@calyxinstitute.org>
Date: Wed, 6 Sep 2023 10:30:51 +0000
Subject: fixup! Bromite subresource adblocker

---
 chrome/common/pref_names.h | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/chrome/common/pref_names.h b/chrome/common/pref_names.h
--- a/chrome/common/pref_names.h
+++ b/chrome/common/pref_names.h
@@ -21,7 +21,6 @@
 
 namespace prefs {
 
-<<<<<<< HEAD
 // *************** PROFILE PREFS ***************
 // These are attached to the user profile
 
@@ -124,25 +123,6 @@ inline constexpr char kImportantSitesDialogHistory[] = "important_sites_dialog";
 // This is the profile creation time.
 inline constexpr char kProfileCreationTime[] = "profile.creation_time";
 
-=======
-// Profile prefs. Please add Local State prefs below instead.
-extern const char kPreinstalledApps[];
-extern const char kSafeBrowsingForTrustedSourcesEnabled[];
-extern const char kDisableScreenshots[];
-extern const char kDownloadRestrictions[];
-extern const char kDownloadBubbleEnabled[];
-extern const char kDownloadBubbleIphSuppression[];
-extern const char kDownloadBubblePartialViewEnabled[];
-extern const char kDownloadBubblePartialViewImpressions[];
-extern const char kDownloadDuplicateFilePromptEnabled[];
-extern const char kForceEphemeralProfiles[];
-extern const char kHomePageIsNewTabPage[];
-extern const char kHomePage[];
-extern const char kAdBlockFiltersURL[];
-extern const char kHttpsOnlyModeEnabled[];
-extern const char kImportantSitesDialogHistory[];
-extern const char kProfileCreationTime[];
->>>>>>> 2b0667e386f10 (Bromite subresource adblocker)
 #if BUILDFLAG(IS_WIN)
 // This is a timestamp of the last time this profile was reset by a third party
 // tool. On Windows, a third party tool may set a registry value that will be
-- 

