From: Tommy Webb <tommy@calyxinstitute.org>
Date: Wed, 6 Sep 2023 12:59:47 +0000
Subject: fixup! Build workarounds for unused functions

---
 chrome/browser/android/tab_android.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/chrome/browser/android/tab_android.cc b/chrome/browser/android/tab_android.cc
--- a/chrome/browser/android/tab_android.cc
+++ b/chrome/browser/android/tab_android.cc
@@ -19,6 +19,7 @@
 #include "base/metrics/user_metrics.h"
 #include "base/trace_event/trace_event.h"
 #include "cc/slim/layer.h"
+#pragma clang diagnostic ignored "-Wunused-function"
 #include "chrome/android/chrome_jni_headers/TabImpl_jni.h"
 #include "chrome/android/chrome_jni_headers/TabUtils_jni.h"
 #include "chrome/browser/android/background_tab_manager.h"
-- 

