From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 5 Oct 2023 16:47:30 +0000
Subject: fixup! Add lifetime options for permissions

---
 components/permissions/permission_request.cc | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/components/permissions/permission_request.cc b/components/permissions/permission_request.cc
--- a/components/permissions/permission_request.cc
+++ b/components/permissions/permission_request.cc
@@ -36,14 +36,10 @@ PermissionRequest::PermissionRequest(
       delete_callback_(std::move(delete_callback)) {}
 
 PermissionRequest::PermissionRequest(
-    const GURL& requesting_origin,
-    RequestType request_type,
-    bool has_gesture,
+    PermissionRequestData request_data,
     PermissionDecidedCallbackWithLifetime permission_decided_callback,
     base::OnceClosure delete_callback)
-    : requesting_origin_(requesting_origin),
-      request_type_(request_type),
-      has_gesture_(has_gesture),
+    : data_(std::move(request_data)),
       permission_decided_callback_withlifetime_(std::move(permission_decided_callback)),
       delete_callback_(std::move(delete_callback)) {}
 
-- 

