From: Tommy Webb <tommy@calyxinstitute.org>
Date: Wed, 6 Sep 2023 09:30:44 +0000
Subject: fixup! Disable conversion measurement api

---
 .../chromium/content/browser/AttributionOsLevelManager.java    | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/content/public/android/java/src/org/chromium/content/browser/AttributionOsLevelManager.java b/content/public/android/java/src/org/chromium/content/browser/AttributionOsLevelManager.java
--- a/content/public/android/java/src/org/chromium/content/browser/AttributionOsLevelManager.java
+++ b/content/public/android/java/src/org/chromium/content/browser/AttributionOsLevelManager.java
@@ -212,8 +212,7 @@ public class AttributionOsLevelManager {
             AttributionOsLevelManagerJni.get().onMeasurementStateReturned(0);
             return;
         }
-        MeasurementManagerFutures mm =
-                MeasurementManagerFutures.from(ContextUtils.getApplicationContext());
+        MeasurementManagerFutures mm = null;
         if (mm == null) {
             AttributionOsLevelManagerJni.get().onMeasurementStateReturned(0);
             return;
-- 

