From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Wed, 31 Aug 2022 02:10:21 +0530
Subject: gn args fix

---
 chrome/browser/privacy/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/chrome/browser/privacy/BUILD.gn b/chrome/browser/privacy/BUILD.gn
--- a/chrome/browser/privacy/BUILD.gn
+++ b/chrome/browser/privacy/BUILD.gn
@@ -11,6 +11,7 @@ if (is_android) {
       "//chrome/android:*",
       "//chrome/browser/feed/android:*",
       "//chrome/browser/ui/android/omnibox:*",
+      "//chrome/browser/ui/android/signin:*",
     ]
     sources = [
       "java/src/org/chromium/chrome/browser/privacy/secure_dns/SecureDnsBridge.java",
-- 
2.37.2

