From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Tue, 19 Apr 2022 03:23:26 +0530
Subject: Re-add trichrome_chrome_64_32_bundle

---
 chrome/android/BUILD.gn | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -3989,6 +3989,14 @@ if (android_64bit_target_cpu) {
     static_library_provider = ":trichrome_library_64_apk"
   }
 
+  monochrome_or_trichrome_public_bundle_tmpl("trichrome_chrome_64_32_bundle") {
+    bundle_suffix = "6432"
+    is_64_bit_browser = true
+    include_32_bit_webview = true
+    use_trichrome_library = true
+    static_library_provider = ":trichrome_library_64_32_apk"
+  }
+
   if (!skip_secondary_abi_for_cq) {
     monochrome_or_trichrome_public_bundle_tmpl("monochrome_32_public_bundle") {
       bundle_suffix = "32"
-- 
2.38.1

