From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 5 Oct 2023 16:33:11 +0000
Subject: Remove overridden method no longer in interface

---
 .../settings/WebLayerAccessibilitySettingsDelegate.java      | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/weblayer/browser/java/org/chromium/weblayer_private/settings/WebLayerAccessibilitySettingsDelegate.java b/weblayer/browser/java/org/chromium/weblayer_private/settings/WebLayerAccessibilitySettingsDelegate.java
--- a/weblayer/browser/java/org/chromium/weblayer_private/settings/WebLayerAccessibilitySettingsDelegate.java
+++ b/weblayer/browser/java/org/chromium/weblayer_private/settings/WebLayerAccessibilitySettingsDelegate.java
@@ -23,11 +23,6 @@ public class WebLayerAccessibilitySettingsDelegate implements AccessibilitySetti
         return mProfile;
     }
 
-    @Override
-    public BooleanPreferenceDelegate getAccessibilityTabSwitcherDelegate() {
-        return null;
-    }
-
     @Override
     public BooleanPreferenceDelegate getReaderForAccessibilityDelegate() {
         return null;
-- 

