From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 7 Sep 2023 14:45:55 +0000
Subject: Sloppy workaround for text treated as binary

---
 .gitattributes | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.gitattributes b/.gitattributes
--- a/.gitattributes
+++ b/.gitattributes
@@ -7,12 +7,12 @@
 # Force LF checkout for all source files
 *.bin binary
 *.c text eol=lf
-*.cc text eol=lf
+*.cc text eol=lf diff
 *.cpp text eol=lf
 *.csv text eol=lf
 *.grd   text eol=lf
 *.grdp   text eol=lf
-*.gn text eol=lf
+*.gn text eol=lf diff
 *.gni text eol=lf
 *.h text eol=lf
 *.html   text eol=lf
-- 

